connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -206,7 +206,9 @@ protected void iteration() {
             // Maybe commit
             if (!committing && (context.isCommitRequested() || now >= nextCommit)) {
                 commitOffsets(now, false);
-                nextCommit += offsetCommitIntervalMs;
+                if (now >= nextCommit) {
```

Comment:
Maybe this is a little simpler?
```java
nextCommit = now + offsetCommitIntervalMs;
```

Reply:
That changes the system in a different way. So in your suggestion, `commitRequest` will always move the `nextCommit` forward -- but notably not into the _far_ future. 

Vs my original change which only moves `nextCommit` forward if it is in the "past".

Unclear to me after re-reading the [original Kip](https://cwiki.apache.org/confluence/display/KAFKA/KIP-89%3A+Allow+sink+connectors+to+decouple+flush+and+offset+commit) what the intention is for this behavior. 

Both solve the problem we faced. Which behavior do you think is better? Always move `nextCommit` forward on each commit or only move `nextCommit` if it is in the past?

Reply:
I don't think it makes a big difference either way. The intent of the offset commit interval is just to make sure committed offsets don't fall too far behind. It does not need to be a strict schedule. It seemed more intuitive and simpler to me to reset the interval. In any case, we should get rid of this relative tracking of the next commit. If we use an absolute time, then we will not have problems like this in the future.

Reply:
Is there any reason not to accept this suggestion? I will go ahead and push an update to this PR next week if there are no further responses so that we can get this fix into the next release.